app/default/account_routes.py

```diff
@@ -1,67 +1,109 @@
-
 import requests
+from app.default.data import get_all_funds
+from app.default.data import get_all_rounds_for_fund
 from app.default.data import get_applications_for_account
-from app.default.data import get_round_data_fail_gracefully
+from app.models.application_summary import ApplicationSummary
+from config import Config
+from flask import Blueprint
+from flask import current_app
+from flask import g
+from flask import redirect
+from flask import render_template
 from flask import request
 from flask import url_for
 from fsd_utils.authentication.decorators import login_required
 from fsd_utils.locale_selector.get_lang import get_lang
-from app.models.application_summary import ApplicationSummary
-from flask import current_app
-from config import Config
-from flask import Blueprint, g, redirect, render_template
-from app.default.routes import current_datetime_after_given, current_datetime_before_given
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_after_given_iso_string,
+)
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_before_given_iso_string,
+)
 
 
 account_bp = Blueprint("account_routes", __name__, template_folder="templates")
 
+
+def build_application_data_for_display(applications: list[ApplicationSummary]):
+
+    application_data_for_display = {"funds": []}
+
+    count_of_applications_for_visible_rounds = 0
+    all_funds = get_all_funds()
+    if all_funds:
+        for fund in all_funds:
+            fund_id = fund["id"]
+            all_rounds_in_fund = get_all_rounds_for_fund(fund_id)
+            fund_data_for_display = {
+                "fund_data": fund,
+                "rounds": [],
+            }
+            for round in all_rounds_in_fund:
+                round_id = round["id"]
+                past_submission_deadline = (
+                    current_datetime_after_given_iso_string(round["deadline"])
+                )
+                not_yet_open = current_datetime_before_given_iso_string(
+                    round["opens"]
+                )
+                apps_in_this_round = [
+                    app for app in applications if app.round_id == round_id
+                ]
+                if not not_yet_open:
```

Comment:
You could get rid of the `# noqa:E501`, double negative, and remove a block of indentation if you make use of the "return early pattern"

```suggestion
                if not_yet_open: continue
                
                for application in apps_in_this_round:
                    ...
```

Reply:
Thanks, good shout